src/main/java/software/amazon/awssdk/http/apache/internal/conn/SdkTlsSocketFactory.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.awssdk.http.apache.internal.conn;
+
+import org.apache.http.HttpHost;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.protocol.HttpContext;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.http.apache.internal.net.SdkSocket;
+import software.amazon.awssdk.http.apache.internal.net.SdkSslSocket;
+import software.amazon.awssdk.utils.Logger;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSocket;
+
+/**
+ * Used to enforce the preferred TLS protocol during SSL handshake.
+ *
+ * Modified by Greengrass to add ALPN header.
+ */
+@SdkInternalApi
+public class SdkTlsSocketFactory extends SSLConnectionSocketFactory {
+
+    private static final Logger log = Logger.loggerFor(SdkTlsSocketFactory.class);
+    private final SSLContext sslContext;
+
+    public SdkTlsSocketFactory(final SSLContext sslContext, final HostnameVerifier hostnameVerifier) {
+        super(sslContext, hostnameVerifier);
+        if (sslContext == null) {
+            throw new IllegalArgumentException(
+                    "sslContext must not be null. " + "Use SSLContext.getDefault() if you are unsure.");
+        }
+        this.sslContext = sslContext;
+    }
+
+    /**
+     * {@inheritDoc} Used to enforce the preferred TLS protocol during SSL handshake.
+     */
+    @Override
+    protected final void prepareSocket(final SSLSocket socket) {
+        // BEGIN GG MODIFICATIONS
+        SSLParameters params = new SSLParameters();
+        params.setApplicationProtocols(new String[]{"x-amzn-http-ca", "http/1.1"});
+        socket.setSSLParameters(params);
```

Comment:
So are you saying these modifications are needed to connect to port 443? How do existing customers connect to port 443 using the AWS SDK then?

Reply:
These are only needed because of IoT Core which requires that HTTPS connections use ALPN when connecting to port 443 instead of 8443.